docs/source/project-configuration.mdx

```diff
@@ -0,0 +1,302 @@
+
+> To learn more about GraphQL operations, check out [Defining operations](./fetching/fetching-data#defining-operations).
+
+#### Schema types
+
+All GraphQL operations depend upon the types in their corresponding GraphQL schema. Your generated operation models also depend on information from the schema. Apollo iOS generates a collection of types that provide the necessary properties and metadata about the types in your GraphQL schema (ie. objects, interfaces, unions, enums, input objects, etc.). These shared schema types are referenced by your generated operation models to provide type information without duplicated code.
+
+#### Test mocks
+
+Apollo iOS provides the ability to generate test mocks objects for use in your test targets. The mocks enable creation of operation models for your tests.
+
+For more information on setting up and using test mocks, see our [test mocks documentation](./testing/test-mocks)
+
+## Configuring your project
+
+When including Apollo iOS in a new project, it's important to decide how you will structure your project, and where your generated GraphQL models fit in that project. Your GraphQL models can be included in your main application target, spread out across many modules, or exposed as an independent module that can be used by multiple applications. Depending on the needs and constraints of your project, you will need to configure the code generation engine and link to the Apollo iOS libraries differently.
+
+There are three primary decisions to be made about your project structure:
+
+> For more specific customization, read about the configuration options below.
+
+### Single target vs multi-module
+
+> ** 1. Will your application consist of a single application target or multiple modules?**
+
+Depending on the size and complexity of your project, it may be built as a single, monolithic application target, or be composed of multiple modules.
+
+For a monolithic application you can simply link your application target to the `Apollo` library. You may optionally, link to `ApolloSQLite` and/or `ApolloWebSocket` to enable their functionality.
+
+To include the Apollo iOS SDK in a project with multiple modules:
+
+- Link `Apollo` to the modules that configure or use the networking and caching APIs.
+    - Optionally, link `ApolloSQLite` and/or `ApolloWebSocket` to the module that sets up your `ApolloClient` to enable their functionality.
+- Link `ApolloAPI` to the modules that only include generated models.
+
+You may also optionally, link `ApolloTestSupport` to your unit test target to create mocks of your generated models.
+
+> For more information about the libraries that make up the Apollo iOS SDK, see the [Apollo iOS SDK components section](#apollo-ios-sdk-components).
+
+> #### **Important: `ApolloCodegenLib` should not be linked to your application targets.**
+> It only supports macOS and should only be linked to by development tools that want to use the Apollo code generation engine.
```

Comment:
Thanks for calling this out explicitly. Hopefully people actually heed the warning.